src/theme/mod.rs

```diff
@@ -46,7 +46,7 @@ pub struct Theme {
 }
 
 impl Theme {
-    pub fn new(src: Option<&PathBuf>) -> Self {
+    pub fn new(src: Option<&Path>) -> Self {
```

Comment:
I was thinking about it but but there is this problem that while in the context of `HtmlConfig` the `theme` is guaranteed, it is not in the context of `MdBook` as the `HtmlConfig` is optional itself

```rust
    pub fn get_theme_path(&self) -> Option<&Path> {
        if let Some(htmlconfig) = self.config.get_html_config() {
            return Some(htmlconfig.get_theme());
        }

        None
    }
```

So removing this option will make us jump through hoops here:
```rust
  --> src/renderer/html_handlebars/hbs_renderer.rs:36:39
   |
36 |         let theme = theme::Theme::new(book.get_theme_path());
```